src/main/java/de/dafuqs/spectrum/status_effects/ImmunityStatusEffect.java

```diff
@@ -19,13 +22,28 @@ public boolean canApplyUpdateEffect(int duration, int amplifier) {
 	@Override
 	public void applyUpdateEffect(LivingEntity entity, int amplifier) {
 		super.applyUpdateEffect(entity, amplifier);
-		WhispyCircletItem.removeNegativeStatusEffects(entity);
+		removeNegativeStatusEffects(entity);
 	}
 
 	@Override
 	public void onApplied(LivingEntity entity, AttributeContainer attributes, int amplifier) {
 		super.onApplied(entity, attributes, amplifier);
-		WhispyCircletItem.removeNegativeStatusEffects(entity);
+		removeNegativeStatusEffects(entity);
+	}
+
+	public static void removeNegativeStatusEffects(@NotNull LivingEntity entity) {
+		Set<StatusEffect> effectsToRemove = new HashSet<>();
+		Collection<StatusEffectInstance> currentEffects = entity.getStatusEffects();
+		for (StatusEffectInstance instance : currentEffects) {
+			StatusEffect effectType = instance.getEffectType();
+			if (effectType.getCategory() == StatusEffectCategory.HARMFUL && !SpectrumStatusEffectTags.bypassesImmunity(effectType)) {
+				effectsToRemove.add(instance.getEffectType());
```

Comment:
nit: why do you use `instance.getEffectType()` here instead of the used-above `effectType`?

Reply:
oops, missed that while bringing it over from WhispyCircletItem